### PR TITLE
Issue #729 Adding a helper function to split command string

### PR DESCRIPTION
**Pull request by hferentschik**
Addresses issue #729 

Pretty much as suggested in issue. Added some more comments and additional unit test.

### CODE REVIEW

File: pkg/util/cmd/split_args.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	splitWithQuotes := func(c rune) bool {
+		switch {
+		// don't split if the character matches the "last matched quote"
+		case c == lastQuote:
+			if consecutiveEscapeCount%2 == 0 {
+				lastQuote = rune(0)
+			}
+			consecutiveEscapeCount = 0
+			return false
+		// found an escapce character, don't split
```
Comment (NickLaMuro) [inline pkg/util/cmd/split_args.go]: Typo:  "escape"

File: pkg/util/cmd/split_args.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+		// don't split if we are still in a quoted string
+		case lastQuote != rune(0):
+			consecutiveEscapeCount = 0
+			return false
+		// don't split if the character is a quote type rune, and set the lastQuote value to that
+		case unicode.In(c, unicode.Quotation_Mark):
+			lastQuote = c
+			consecutiveEscapeCount = 0
+			return false
+		// split the string if it is a space
```
Comment (NickLaMuro) [inline pkg/util/cmd/split_args.go]: Not sure if it is worth noting in this comment, but maybe include something along the lines of "... and none of the other conditionals are satisfied".   That said, these are all very concise comments, so maybe leave it as is if that was the goal (I myself tend to get a little lengthy... character flaw...).

File: pkg/util/cmd/split_args_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	if args[3] != "01/01/1900" {
+		t.Errorf("Expected the second arg to be '01/01/1900', but was '%s'", args[2])
+	}
+
+	if args[4] != "+%s" {
+		t.Errorf("Expected the second arg to be '+%%s', but was '%s'", args[2])
+	}
+}
+
+func TestProperlyHandlesNestedSingleQuotesWithinDoubleQuotes(t *testing.T) {
```
Comment (budhrg) [inline pkg/util/cmd/split_args_test.go]: :+1:

File: pkg/util/cmd/split_args_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	if args[3] != "01/01/1900" {
+		t.Errorf("Expected the second arg to be '01/01/1900', but was '%s'", args[2])
+	}
+
+	if args[4] != "+%s" {
+		t.Errorf("Expected the second arg to be '+%%s', but was '%s'", args[2])
+	}
+}
+
+func TestProperlyHandlesNestedSingleQuotesInNestedDoubleQuotes(t *testing.T) {
```
Comment (budhrg) [inline pkg/util/cmd/split_args_test.go]: :+1:

File: pkg/util/cmd/split_args_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+	if args[3] != "01/01/1900" {
+		t.Errorf("Expected the second arg to be '01/01/1900', but was '%s'", args[2])
+	}
+
+	if args[4] != "+%s" {
+		t.Errorf("Expected the second arg to be '+%%s', but was '%s'", args[2])
+	}
+}
+
+func TestProperlyHandlesEscapedDoubleQuotesWithinDoubleQuotes(t *testing.T) {
```
Comment (budhrg) [inline pkg/util/cmd/split_args_test.go]: :+1:

File: pkg/util/cmd/split_args_test.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+func TestReturnsAnArrayOfStrings(t *testing.T) {
+	args := SplitCmdString(`date 1423`)
+
+	if args[0] != "date" {
+		t.Errorf("Expected the first arg to be 'date', but was '%s'", args[0])
+	}
+
+	if args[1] != "1423" {
+		t.Errorf("Expected the second arg to be '1423', but was '%s'", args[1])
+	}
```
Comment (praveenkumar) [inline pkg/util/cmd/split_args_test.go]: Not a blocker but we can do this like and can be applied for each test case.

```
expectedArgsList := []string{"date", "1423"}
for i, v := range expectedArgsList {
		if args[i] != v {
			t.Errorf("Expected the %d arg to be %s, but Got '%s'", (i+1), v, args[i])
		}
	}
```

File: pkg/util/cmd/split_args_test.go [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+func TestProperlyHandlesWhitespaceWithinSingleQuotes(t *testing.T) {
+	args := SplitCmdString(`date -f '%a %b %d %T %Z %Y' "01/01/1900" '+%s'`)
```
Comment (budhrg) [inline pkg/util/cmd/split_args_test.go]: If we use approach suggested here https://github.com/minishift/minishift/pull/764/files/14f24adb92d3ed3f95fc78c4c2f0dbee7d5b7d2f#r112129096, then I believe, this test case would be much smaller.

**Comment by budhrg** [comment]
@hferentschik , here the reference to issue id is wrong it should be https://github.com/minishift/minishift/issues/729

**Comment by hferentschik** [comment]
> here the reference to issue id is wrong it should be #221

thanks, will fix

File: pkg/util/cmd/split_args.go (hunk 1 above)
Comment (hferentschik) [inline pkg/util/cmd/split_args.go]: thanks :-)

File: pkg/util/cmd/split_args.go (hunk 2 above)
Comment (hferentschik) [inline pkg/util/cmd/split_args.go]: added it to be explicit

**Pull request merged**